packages/alfa-rules/src/sia-r83/rule.ts

```diff
@@ -328,33 +333,42 @@ namespace ClippingAncestor {
         return Overflow.Handle;
       }
 
-      const horizontalOverflow = overflow(element, device, "x");
-      let result = horizontalOverflow;
+      let horizontalOverflow = overflow(element, device, "x");
       if (horizontalOverflow === Overflow.Clip) {
-        // We should check here whether the element's size is constrained
-        // horizontally or not. We assume all elements are because the pages
-        // usually don't scroll infinitely in the horizontal dimension.
-        // This mostly gives us the wrong clipper.
-        result =
+        if (
           inSameBlock &&
           hasUsedStyle(
             "text-overflow",
             (value) => value.is("ellipsis"),
             device,
           )(element)
-            ? // If we are still in the same block, `text-overflow` can handle it
-              // This is where inline elements will not have a `text-overflow` and
-              // return `Clip` even if their width is not constrained!
-              Overflow.Handle
-            : Overflow.Clip;
+        ) {
+          // As long as we are in the same block, we consider ellispis as an
+          // indication that clipping was taken into consideration.
+          horizontalOverflow = Overflow.Handle;
+        }
+
+        if (
+          constrainingAncestor(element, device, "width").every(
+            isTwiceAsBig(element, device)("width"),
+          )
+        ) {
+          // If the element is not itself constrained, and twice as small as its
+          // closest constraining ancestor, it has room to grow.
+          // We return a somewhat incorrect value here as it will ultimately
+          // turn into an Oatcome.WrapsText while a Outcome.IsContainer would
+          // be more correct. This is OK since we do not really rely on that
+          // information anywhere.
+          horizontalOverflow = Overflow.Handle;
+        }
```

Comment:
@mvy-siteimprove this should fix https://github.com/Siteimprove/dory/pull/330#issuecomment-2404291123